packages/tools/devtools/devtools-core/src/data-visualization/VisualTree.ts

```diff
@@ -68,6 +68,21 @@ export interface VisualNodeBase {
 	 * Consumers of this value should attempt to handle unrecognized values gracefully.
 	 */
 	nodeKind: VisualNodeKind | string;
+
+	/**
+	 * The "kind" of tree node.
+	 *
+	 * @remarks
+	 *
+	 * This will generally be {@link VisualNodeKind}, but for forwards/backwards compatability reasons,
+	 * we allow this to be an arbitrary string.
+	 * Consumers of this value should attempt to handle unrecognized values gracefully.
+	 */
+
+	/**
+	 * (optional) The types of edits allowed
```

Comment:
We should expand on this a bit more. Especially, we should note that this object being present indicates that the corresponding data is editable, and its absence indicates that it is not.

Reply:
I expanded on it, check the new commit